File: backend/src/main/java/it/prevt/backend/entity/Prospect.java
package it.prevt.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.UuidGenerator;

import java.time.OffsetDateTime;
import java.util.UUID;


@Getter @Setter
@NoArgsConstructor @AllArgsConstructor @Builder
@Entity
@Table(name = "prospect")
public class Prospect {

    @Id
    @GeneratedValue
    @UuidGenerator
    @Column(nullable = false, updatable = false)
    private UUID id;

    @ManyToOne(optional = false, fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "ragione_sociale", nullable = false)
    private String ragioneSociale;

    @Column(name = "partita_iva", nullable = false)
    private String partitaIva;

    @Column(name = "codice_fiscale")
    private String codiceFiscale;

    private String indirizzo;
    private String citta;
    private String cap;
    private String provincia;
    private String telefono;
    private String email;

    @Column(nullable = false)
    private String tipo = "prospect";

    @Column(name = "tipo_prospect")
    private String tipoProspect = "Professional";

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;
}

File: frontend/src/types/prospect.ts
import {ISODateString} from "@/types/index.ts";

export interface ProspectBean {
    id: string;
    userId?: string;
    ragioneSociale: string;
    partitaIva: string;
    codiceFiscale?: string;
    indirizzo: string;
    citta: string;
    cap: string;
    provincia: string;
    telefono?: string;
    email?: string;
    tipo: 'prospect' | 'cliente';
    tipoProspect?: 'Professional' | 'Finale';
    scontoCliente: number;
    createdAt?: ISODateString;
    updatedAt?: ISODateString;
}
